pipeline/id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/steps/SourceIdentifierScanner.scala

```diff
@@ -0,0 +1,108 @@
+package uk.ac.wellcome.platform.idminter.steps
+
+import grizzled.slf4j.Logging
+import io.circe.Json
+import io.circe.optics.JsonPath.root
+import io.circe.optics.JsonOptics._
+import monocle.function.Plated
+import uk.ac.wellcome.models.Implicits._
+import uk.ac.wellcome.models.work.internal.SourceIdentifier
+import uk.ac.wellcome.platform.idminter.models.Identifier
+
+import scala.annotation.tailrec
+import scala.util.{Failure, Success, Try}
+
+/**
+  * SourceIdentifierScanner provides 2 methods:
+  *
+  * - `scan` takes Json and returns all of the sourceIdentifiers that are in it
+  * - `update` takes Json and a map of (SourceIdentifier -> Identifier) and adds
+  *   a canonicalId field next to sourceIdentifiers, as well as replacing
+  *   `identifiedType` fields with `type` fields of the same value.
```

Comment:
tiny nit: listing the methods in the order they're described would have been easier to scan

Reply:
Also, what is a Scanner doing updating JSON?